httpx/utils.py

```diff
@@ -64,3 +66,12 @@ def guess_json_utf(data: bytes) -> typing.Optional[str]:
             return "utf-32-le"
         # Did not detect a valid UTF-32 ascii-range character
     return None
+
+
+def get_netrc_login(url: str) -> typing.Union[typing.Tuple[str, str, str], None]:
```

Comment:
Signature should either be `host` or `authority` here, right?
It's not clear to me from a glance at the netrc docs, which of those two is actually the correct thing to use?
Also we should prefer `typing.Optional[typing.Tuple[str, str, str]]` as the return annotation.

Reply:
`machine` in the config. I changed it as host I can change again if necessary